server/rest/src/main/java/org/infinispan/rest/ServerSentEvent.java

```diff
@@ -0,0 +1,151 @@
+package org.infinispan.rest;
+
+import java.nio.charset.StandardCharsets;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.DecoderResult;
+import io.netty.handler.codec.http.HttpContent;
+
+public class ServerSentEvent implements HttpContent {
+   private static final byte[] EVENT = "event: ".getBytes();
+   private static final byte[] NL = "\n".getBytes();
+   private static final byte[] ID = "id: ".getBytes();
+   private static final byte[] DATA = "data: ".getBytes();
+
+   private final String id;
+   private final String event;
+   private final String data;
+   private DecoderResult decoderResult;
+
+   public ServerSentEvent(String id, String event, String data) {
+      this.id = id;
+      this.event = event;
+      this.data = data;
+   }
+
+   public ServerSentEvent(String event, String data) {
```

Comment:
Only this constructor is used, so `id` is always `null`. Do you have plans to use the ID field? If not, I think we should remove the dead code until it is required.

Reply:
Removed all the dead code.